### PR TITLE
separate unit tests from e2etest

### PR DESCRIPTION
**Pull request by GregHanson**
**What this PR does / why we need it**:
Separates `make test` test case from `make e2etest` to speed up circleci jobs by running them in parallel

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,53 +1,33 @@
 version: 2
 
 jobs:
-  build:
+  e2etest:
```
Comment (rshriram) [inline .circleci/config.yml]: pilot-integration

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,53 +1,33 @@
...
-      - run:
-          command: |
-            ln -s ~/.kube/config pilot/platform/kube/
-      - restore_cache:
-          keys:
-            - bazel-cache-{{ checksum "WORKSPACE" }}
-      - run: make test
-      - save_cache:
-          key: bazel-cache-{{ checksum "WORKSPACE" }}
-          paths:
```
Comment (rshriram) [inline .circleci/config.yml]: why remove the cache?

File: .circleci/config.yml [hunk 3]
```diff
@@ -57,9 +37,28 @@ jobs:
       - run: cd /go/src/istio.io/istio; dep ensure
       - run: cd /go/src/istio.io/istio; go build -i ./pilot/... ./mixer/... ./broker/...
```
Comment (rshriram) [inline .circleci/config.yml]: take a look at setup remote docker. it allows you to build docker images
you can build and then push docker images and then you can start up the e2e tests, while you can run unit tests simultaneously

File: .circleci/config.yml [hunk 4]
```diff
@@ -57,9 +37,28 @@ jobs:
...
+    docker:
+      - image: circleci/golang:1.9
+
+      - image: gcr.io/google_containers/kube-apiserver-amd64:v1.7.3
+        name: istio-apiserver
+        command: ["kube-apiserver", "--etcd-servers", "http://etcd:2379", "--service-cluster-ip-range", "10.99.0.0/16", "--insecure-port", "8080", "-v", "2", "--insecure-bind-address", "0.0.0.0"]
+
+      - image: quay.io/coreos/etcd:latest
+        name: etcd
+        command: ["/usr/local/bin/etcd", "-advertise-client-urls=http://0.0.0.0:2379", "-listen-client-urls=http://0.0.0.0:2379"]
```
Comment (rshriram) [inline .circleci/config.yml]: nice work

File: .circleci/config.yml [hunk 5]
```diff
@@ -57,9 +37,28 @@ jobs:
...
+    steps:
+      - checkout
+      - run: pilot/bin/install-bazel.sh
+      - run: ln -s /go/src/istio.io/istio/install/consul/kubeconfig /go/src/istio.io/istio/pilot/platform/kube/config
```
Comment (rshriram) [inline .circleci/config.yml]: just cp the file

File: .circleci/config.yml [hunk 6]
```diff
@@ -1,65 +1,72 @@
...
+    environment:
+      CHANGE_MINIKUBE_NONE_USER: true
     steps:
       - checkout
+      - run: curl -Lo kubectl https://storage.googleapis.com/kubernetes-release/release/v1.7.4/bin/linux/amd64/kubectl && chmod +x kubectl && sudo mv kubectl /usr/local/bin/
+      - run: curl -Lo minikube https://storage.googleapis.com/minikube/releases/v0.22.3/minikube-linux-amd64 && chmod +x minikube && sudo mv minikube /usr/local/bin/
+      - run:
+          command: |
+            sudo -E minikube start \
+            --extra-config=apiserver.Admission.PluginNames="Initializers,NamespaceLifecycle,LimitRanger,ServiceAccount,DefaultStorageClass,GenericAdmissionWebhook,ResourceQuota" \
```
Comment (cmluciano) [inline .circleci/config.yml]: Nit: might want to add `--extra-config=apiserver.Authorization.Mode=RBAC \` to make sure that all default RBAC rules are setup correctly (at least until it is enabled by default)

**Pull request merged**